src/components/input/input.tsx

```diff
@@ -720,6 +722,23 @@ export class Input implements LabelableComponent, FormComponent, InteractiveComp
     }
   }
 
+  hiddenInputChangeHandler = (event: Event): void => {
+    if ((event.target as HTMLInputElement).name === this.name) {
```

Comment:
Code for this should be moved into a util module to make it easier for components to apply. This can be done separately.

Reply:
Created a new issue #5171